prime-router/src/test/kotlin/fhirengine/engine/fhirRouterTests/RoutingTests.kt

```diff
@@ -623,8 +641,27 @@ class RoutingTests {
             "(%performerState.exists() and %performerState = 'CA') or (%patientState.exists() " +
                 "and %patientState = 'CA')"
         )
-
+        val result = engine.evaluateFilterCondition(
+            filter,
+            bundle,
+            false
+        )
         // act & assert
-        assertThat(engine.evaluateFilterCondition(filter, bundle, false)).isTrue()
+        assertThat(result).isTrue()
+    }
+
+    @Test
+    fun `test logFilterResults`() {
+        val fhirData = File("src/test/resources/fhirengine/engine/routerDefaults/qual_test_0.fhir").readText()
+        val bundle = FhirTranscoder.decode(fhirData)
+        val report = Report(one, listOf(listOf("1", "2")), TestSource, metadata = UnitTestUtils.simpleMetadata)
+        val settings = FileSettings().loadOrganizations(oneOrganization)
+        var qualFilter = listOf("Bundle.entry.resource.ofType(Provenance).count() > 0")
+
+        val engine = spyk(makeFhirEngine(metadata, settings, TaskAction.route) as FHIRRouter)
+
+        assertThat(report.filteringResults.count()).isEqualTo(0)
+        engine.logFilterResults(qualFilter, bundle, report, receiver, ReportStreamFilterType.QUALITY_FILTER)
+        assertThat(report.filteringResults.count()).isGreaterThan(0)
```

Comment:
Add an assertThat that actually looks at what is inside the filteringResults value and verifies the correct (and useful) data is present.

Reply:
Makes sense. done!